optimum/neuron/pipelines/transformers/base.py

```diff
@@ -0,0 +1,238 @@
+    subfolder: str = "",
+    use_auth_token: Optional[Union[bool, str]] = None,
+    revision: str = "main",
+    model_kwargs: Optional[Dict[str, Any]] = None,
+    config: AutoConfig = None,
+    **kwargs,
+):
+    if model_kwargs is None:
+        model_kwargs = {}
+
+    # loads default model
+    if model is None:
+        model_id = SUPPORTED_TASKS[targeted_task]["default"]
+        if input_shapes is None:
+            input_shapes = {"batch_size": 1, "sequence_length": 128}
+            logger.warning(f"No input shapes provided, using default shapes, {input_shapes}")
+        model = SUPPORTED_TASKS[targeted_task]["class"][0].from_pretrained(model_id, export=True, **input_shapes)
+    # loads model from model id and converts it to neuronx optionally
+    elif isinstance(model, str):
+        model_id = model
+        neuronx_model_class = SUPPORTED_TASKS[targeted_task]["class"][0]
+        model = neuronx_model_class.from_pretrained(model, export=export, **model_kwargs, **input_shapes)
+    # uses neuron model
+    elif isinstance(model, NeuronBaseModel):
+        if tokenizer is None and load_tokenizer:
+            for preprocessor in model.preprocessors:
+                if isinstance(preprocessor, (PreTrainedTokenizer, PreTrainedTokenizerFast)):
+                    tokenizer = preprocessor
+                    break
+            if tokenizer is None:
+                raise ValueError(
+                    "Could not automatically find a tokenizer for the NeuronBaseModel, you must pass a tokenizer explictly"
+                )
+        if feature_extractor is None and load_feature_extractor:
+            for preprocessor in model.preprocessors:
+                if isinstance(preprocessor, SequenceFeatureExtractor):
+                    feature_extractor = preprocessor
+                    break
+            if feature_extractor is None:
+                raise ValueError(
+                    "Could not automatically find a feature extractor for the ORTModel, you must pass a "
+                    "feature_extractor explictly"
+                )
+        model_id = None
+    else:
+        raise ValueError(
+            f"""Model {model} is not supported. Please provide a valid model either as string or NeuroModel.
+            You can also provide non model then a default one will be used"""
+        )
+    return model, model_id, tokenizer, feature_extractor
+
+
+def pipeline(
+    task: str = None,
+    model: Optional[Any] = None,
+    tokenizer: Optional[Union[str, PreTrainedTokenizer]] = None,
+    feature_extractor: Optional[Union[str, PreTrainedFeatureExtractor]] = None,
+    use_fast: bool = True,
+    export: bool = False,
+    input_shapes: Optional[Dict[str, int]] = None,
+    use_auth_token: Optional[Union[str, bool]] = None,
+    revision: Optional[str] = None,
+    trust_remote_code: Optional[bool] = None,
+    *model_kwargs,
+    **kwargs,
+) -> Pipeline:
+    if task not in list(NEURONX_SUPPORTED_TASKS.keys()):
```

Comment:
Apparently `task` cannot be None?
Why setting a default value to `None` here?

Reply:
i guess for name args. Copied from optimum